app.js

```diff
@@ -35,15 +44,16 @@ const app = express();
 app.set('port', process.env.PORT || 3000);
 app.set('views', path.join(__dirname, 'views'));
 app.set('view engine', 'pug');
-app.use(express.static(path.join(__dirname, 'public')));
 
 /*
  * Use application-level middleware for common functionality, including logging, parsing, and session handling.
  */
-app.use(require('morgan')('combined'));
-app.use(require('cookie-parser')());
-app.use(require('body-parser').urlencoded({ extended: true }));
-app.use(require('express-session')({ secret: 'keyboard', resave: false, saveUninitialized: false }));
+app.use(express.static(path.join(__dirname, 'public')));
+app.use(bodyParser.json()); // support json encoded bodies
+app.use(bodyParser.urlencoded({ extended: true })); // support encoded bodies
+app.use(morgan('combined'));
+app.use(cookie());
+app.use(session({ secret: 'keyboard', resave: false, saveUninitialized: false }));
 
```

Comment:
Not really sure what these things do, but if you are confident about them they look fine. Are these all things that need to be done in app.js?